### DISCUSSION TITLE
Separate windows (remotesessions) instead of tabs in mainwindow of Rustdesk Client

### DISCUSSION THREAD

**Discussion by csch0en** [rustdesk/rustdesk#2889] [Q&A] [answered]
I Love Rustdesk.

The only thing to improve for me is:
It would be a nice feature if you could display each session in its own window and not in the same Rustdesk window as tabs. (e.g. for Monitoring of several computers at the same time on one monitor).
BR.

**Comment by GwarlekMeltphace** [comment]
I noticed this when trying the Nightly build.  1.1.9 still has separate windows.
If this goes to the next stable version, it would kill how I work, with one separate RustDesk session on a second monitor. 

Sidenote for anyone searching for this issue: 
I installed the Nightly build to fix the cursor not changing. To fix this in Windows 10, go to Settings > Ease of Access > Mouse and turn on Mouse Keys.  The cause of this was someone at work unplugged my mouse! Found the fix on another remote clients forum.

**Comment by fernandeusto** [comment]
This would also be a good idea when the client of each tab has different screen resolutions, it is easier to use a window for each client than the use of tabs as in the nightly version.

**Comment by SonDG1974** [comment]
It's a necessary function
It would be really convenient to compare each client
I'd like to have an option for tab or window separation

**Comment by pixelstuff** [comment]
Tear off / merge-able tabs like most browsers have now days would be ideal.

**Comment by t3ramos** [comment]
any comment from the devs? been using the sciter version because there we have multiple windows for sessions.

**Comment by PLN** [comment]
Adding my +1 to this issue. I was very disappointed to realise multiple connections are now tabbed in a single window when going from 1.1.9 to nightly. I vastly prefer separate server and client windows, and would like even separate processes, so one crash/kill doesn't take out the rest. While I like a lot of other features, and was actually looking forward to switching to RustDesk as my main VNC replacement, going single-window would make it a non-starter for me.

Edit: A thought occurred that an easy fix might be for the command line to accept a parameter to force a new process instance, rather than joining the existing one. Something along the line of: RustDesk.exe --connect 12345 --forceinstance

**Comment by omarpiani** [comment]
with linux client you can click open new windows and have it, in windows I found no trick :(

**Comment by rustdesk** [comment, marked as answer]
Seperate window mode will be added in next version, 1.2.2.  Default tab mode won't be changed, no way to make every one happy in this world.

**Reply by lemmyreddit** [reply]
how to enable the separate window mode?

**Reply by bilogic** [reply]
yes, how to separate the tabs?

**Reply by fufesou** [reply]
![1722848033385](https://github.com/user-attachments/assets/c2f4b0c0-8317-414e-ac50-3303577c9911)


![1722848105037](https://github.com/user-attachments/assets/028eddb8-5677-45e9-a8ba-9bcaae58924a)

**Comment by pldtweb** [comment]
I have multiple desktops connected, how can I see them side by side?

**Reply by dinger1986** [reply]
Did you read the comment above? It will be in 1.2.2

**Reply by lemmyreddit** [reply]
version 1.2.2 nightly was released today, how to enable the separate window mode?

**Reply by deep-soft** [reply]
1.2.2, as you said, is nightly; separate windows mode is WIP (Work In Progress).

**Comment by dinger1986** [comment]
Can I take this a little further and say longer term it would be cool to have a way to monitor screens and then double click to make them larger, kind of like a cctv screen but maybe that's a side project for someone

**Reply by pldtweb** [reply]
will be great ! I have connected 3 computers via rustdesk but I am not able to look them at the same time , i wish the 1.2.2 gets released really soon

**Comment by mikekgr** [comment]
There's a problem in the separate windows implementation. If you for example, open separately two remote connections, the local keyboard it is functional ONLY in one of them. Please check it I suppose that it's very easy to verify it. If you deselect separate remote window from settings then you have keyboard in both of the remotes as it should...
By the way, congratulations it's VERY good to have separate remote window (I think should be say: Separate Remote Windows , in plural)

**Reply by rustdesk** [reply]
@mikekgr  Post it on PR. The dev may not see here.

**Reply by mikekgr** [reply]
Sorry for my mistake. I will follow your suggestion next time as I saw that you have already pass it to devs

**Reply by dignow** [reply]
https://github.com/rustdesk/rustdesk/pull/5267

**Comment by chris-31337** [comment]
The "Seperate window mode" is fantastic! It would be even greater if there was a key modifier (e.g. holding ctrl or shift during click on monitor) to open a separate window on demand, rather than (only) having a permanent setting. Is this the right place for such a feature suggestion? I love Rustdesk!
